types/artifacts.d.ts

```diff
@@ -213,6 +215,36 @@ declare global {
         requestId: string | null
       }
 
+      /** @see https://sourcemaps.info/spec.html#h.qz3o9nc69um5 */
+      export type RawSourceMap = {
+        /** File version (always the first entry in the object) and must be a positive integer. */
+        version: number
+        /** A list of original sources used by the “mappings” entry. */
+        sources: string[]
+        /** A list of symbol names used by the “mappings” entry. */
+        names: string[]
+        /** An optional source root, useful for relocating source files on a server or removing repeated values in the “sources” entry. This value is prepended to the individual entries in the “source” field. */
```

Comment:
> This value is prepended to the individual entries in the “source” field.

we might want to consider doing this in the gatherer so audits don't have to all deal with this (I believe the source map would still be valid and could just as easily come from the server that way), but we can think about it (parsing will likely be in a computed artifact, so it could happen there)

Reply:
Let's definitely punt on this for now. My first reaction is that the gatherer shouldn't modify the map at all. I don't imagine there's a difference (would want to explore `source-map` and other tools first) but you never know.

Maybe a utility function `getPathForSource` so each audit wouldn't need to? we'll cross this bridge later

Reply:
> Let's definitely punt on this for now. My first reaction is that the gatherer shouldn't modify the map at all.

I'm also ok with leaving as-is for experimentation, but we should be open to changing in a couple of ways to make dealing with artifacts easier. e.g. being able to link up with `ScriptElements` might also be a thing, since audits can't refetch scripts since the driver isn't around by then.